src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs

```diff
@@ -83,7 +83,7 @@ private bool UsesKnownDefaultValue(IOperation value, ITypeSymbol type)
 
             // If this is default(T) or new ValueType(), it's the default.
             if (value is IDefaultValueOperation ||
-                (type.IsValueType && value is IObjectCreationOperation oco && oco.Arguments.IsEmpty && oco.Initializer is null))
+                (type.IsValueType && value is IObjectCreationOperation oco && oco.Arguments.IsEmpty && oco.Initializer is null && oco.Constructor.IsImplicitlyDeclared))
```

Comment:
Per https://github.com/dotnet/roslyn-analyzers/issues/5887#issuecomment-1044617402, the diagnostic should never be reported for structs.

Reply:
My comment was suggesting we shouldn't report it when initializing instance members on structs.